src/instructlab/lab.py

```diff
@@ -310,61 +311,108 @@ def diff(ctx, taxonomy_path, taxonomy_base, yaml_rules, quiet):
     type=click.Path(),
     default=config.DEFAULT_MODEL_PATH,
     show_default=True,
-    help="Path to the model used during generation.",
+    help="Local or HF path to the model used during generation.",
+)
+@click.option(
+    "--log-file",
+    type=click.Path(),
+    help="Log file path to write server logs to.",
 )
 @click.option(
-    "--gpu-layers",
+    "--llama-cpp-gpu-layers",
     type=click.INT,
     help="The number of layers to put on the GPU. The rest will be on the CPU. Defaults to -1 to move all to GPU.",
 )
-@click.option("--num-threads", type=click.INT, help="The number of CPU threads to use.")
+@click.option("--llama-cpp-num-threads", type=click.INT, help="The number of CPU threads to use.")
 @click.option(
-    "--max-ctx-size",
+    "--llama-cpp-max-ctx-size",
     type=click.INT,
     help="The context size is the maximum number of tokens considered by the model, for both the prompt and response. Defaults to 4096.",
 )
 @click.option(
-    "--model-family",
+    "--llama-cpp-model-family",
     type=str,
     help="Force model family to specify which chat template to serve with",
 )
 @click.option(
-    "--log-file",
+    "--use-vllm",
+    type=click.BOOL,
+    default=False,
+    show_default=True,
+    help="Use vllm for model serving.",
+)
+@click.option(
+    "--vllm-chat-template",
     type=click.Path(),
-    help="Log file path to write server logs to.",
+    help="Path to chat template file. Chat templates must be in .jinja format."
+)
+@click.option(
+    "--vllm-args",
+    type=str,
+    help="Specific arguments to pass into vllm. The value passed into this flag must be surrounded by double quotes.\n Example: --vllm-args '--dtype auto --enable-lora'",
 )
 @click.pass_context
-def serve(
-    ctx, model_path, gpu_layers, num_threads, max_ctx_size, model_family, log_file
-):
+def serve(ctx, model_path, log_file, llama_cpp_gpu_layers, llama_cpp_num_threads, llama_cpp_max_ctx_size, llama_cpp_model_family, use_vllm, vllm_chat_template, vllm_args):
     """Start a local server"""
-    # pylint: disable=C0415
-    # Local
-    from .server import ServerException, server
+    from pathlib import PurePath
```

Comment:
Please move this to top-level import. Also why are you using `PurePath` instead of `Path` here? There is rarely a need to use `PurePath` except for special circumstances. See top level module docs of https://docs.python.org/3/library/pathlib.html